src/work-allocation-2/containers/work-case-list/work-case-list.component.spec.ts

```diff
@@ -0,0 +1,459 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { Router } from '@angular/router';
+import { LoadingService, PaginationModule } from '@hmcts/ccd-case-ui-toolkit';
+import { FeatureToggleService } from '@hmcts/rpx-xui-common-lib';
+import { of } from 'rxjs';
+
+import { WorkCaseListComponent } from './work-case-list.component';
+import { FieldConfig, SortField } from '../../models/common';
+import { CaseService, SortOrder } from '../../enums';
+import { Case, CaseAction, CaseServiceConfig } from '../../models/cases';
+import { PaginationParameter } from '../../models/dtos';
+import { WorkAllocationComponentsModule } from '../../components/work-allocation.components.module';
+import { WorkAllocationCaseService } from '../../services';
+import { getMockCases, MockRouter } from '../../tests/utils.spec';
+import { ConfigConstants } from '../../components/constants';
+
+@Component({
+  template: `
+    <exui-work-case-list
+      [fields]='fields'
+      [cases]='cases'
+      [casesTotal]="casesTotal"
+      [caseServiceConfig]="caseServiceConfig"
+      [sortedBy]="CaseSortField"
+      [pagination]="pagination"></exui-work-case-list>`
+})
+class WrapperComponent {
+  @ViewChild(WorkCaseListComponent) public appComponentRef: WorkCaseListComponent;
+  @Input() public fields: FieldConfig[];
+  @Input() public cases: Case[];
+  @Input() public casesTotal: number;
+  @Input() public caseServiceConfig: CaseServiceConfig;
+  @Input() public pagination: PaginationParameter;
+  @Input() public sortedBy: SortField;
+}
+
+/**
+ * Mock cases
+ */
+function getCases(): Case[] {
+  return getMockCases();
+}
+
+/**
+ * Mock fields
+ */
+function getFields(): FieldConfig[] {
+  return ConfigConstants.MyCases;
+}
+
+/**
+ * Mock CaseServiceConfig.
+ */
+function getCaseService(): CaseServiceConfig {
+  return {
+    service: CaseService.IAC,
+    defaultSortDirection: SortOrder.ASC,
+    defaultSortFieldName: 'dueDate',
+    fields: getFields(),
+  };
+}
+
+describe('CaseListComponent', () => {
+  let component: WorkCaseListComponent;
+  let wrapper: WrapperComponent;
+  let fixture: ComponentFixture<WrapperComponent>;
+  let routerSpy: jasmine.SpyObj<any>;
+  const mockRouter: MockRouter = new MockRouter();
+  const mockWorkAllocationService = jasmine.createSpyObj('mockWorkAllocationService', ['getCase']);
+  const mockFeatureToggleService = jasmine.createSpyObj('featureToggleService', ['isEnabled', 'getValue']);
+  const mockLoadingService = jasmine.createSpyObj('mockLoadingService', ['register', 'unregister']);
+  beforeEach((() => {
+    routerSpy = jasmine.createSpyObj('Router', ['navigate']);
+    TestBed.configureTestingModule({
+      imports: [
+        WorkAllocationComponentsModule,
+        CdkTableModule,
+        PaginationModule
+      ],
+      declarations: [WorkCaseListComponent, WrapperComponent],
+      providers: [
+        { provide: WorkAllocationCaseService, useValue: mockWorkAllocationService },
```

Comment:
_Question:_
Is this needed? The constructor for the `WorkCaseListComponent` doesn't accept it as a parameter. The same applies to the `LoadingService` and the `FeatureToggleService`.

Reply:
Removed all three.